File: backend/payment/src/main/java/com/bbmovie/payment/entity/SubscriptionPlan.java
package com.bbmovie.payment.entity;

import com.bbmovie.payment.entity.base.BaseEntity;
import com.bbmovie.payment.entity.enums.PlanType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.Table;
import lombok.*;

import java.math.BigDecimal;

@Getter
@Setter
@Builder
@Entity
@Table(name = "subscription_plans")
@NoArgsConstructor
@AllArgsConstructor
public class SubscriptionPlan extends BaseEntity {

    @Column(name = "name", nullable = false)
    private String name;

    @Column(name = "type", nullable = false)
    private PlanType planType;

    @Column(name = "base_amount", nullable = false)
    private BigDecimal baseAmount;

    @Column(name = "base_currency", nullable = false)
    private String baseCurrency;

    @Enumerated(EnumType.STRING)
    @Column(name = "billing_cycle", nullable = false)
    private com.bbmovie.payment.entity.enums.BillingCycle billingCycle;

    @Column(name = "description")
    private String description;

    @Column(name = "features")
    private String features;

    @Builder.Default
    @Column(name = "active", nullable = false)
    private boolean active = true;

    @Column(name = "monthly_price", nullable = false)
    private BigDecimal monthlyPrice;

    @Column(name = "annual_discount_percent", nullable = false)
    private int annualDiscountPercent; // 0-100

    @Column(name = "student_discount_percent", nullable = false)
    private int studentDiscountPercent; // 0-100
}

File: backend/payment/src/main/kotlin/com/bbmovie/payment/entity/SubscriptionPlan.kt
package com.bbmovie.payment.entity

import com.bbmovie.payment.entity.base.BaseEntity
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Table
import java.math.BigDecimal

@Entity
@Table(name = "subscription_plans")
open class SubscriptionPlan : BaseEntity() {

    @Column(name = "name", nullable = false)
    open lateinit var name: String

    @Column(name = "base_amount", nullable = false)
    open lateinit var baseAmount: BigDecimal

    @Column(name = "base_currency", nullable = false)
    open lateinit var baseCurrency: String

    @Column(name = "billing_cycle", nullable = false)
    open lateinit var billingCycle: String

    @Column(name = "description")
    open var description: String? = null

    @Column(name = "features")
    open var features: String? = null

    @Column(name = "active", nullable = false)
    open var active: Boolean = true
}